tokio-boring/src/lib.rs

```diff
@@ -254,6 +254,19 @@ where
                 return Poll::Pending;
             }
             Err(e) => {
+                // If boring returns PROTOCOL_IS_SHUTDOWN then the connection
+                // has already been shutdown and we can just return Ok(()), as
+                // this was exactly what we wanted to do anyway.
+                if e.code() == ErrorCode::SSL {
+                    if let Some(stack) = e.ssl_error() {
+                        if let Some(first) = stack.errors.first() {
+                            if first.code() as i32 == boring_sys::SSL_R_PROTOCOL_IS_SHUTDOWN {
+                                return Poll::Ready(Ok(()));
```

Comment:
Hmm, could we do this in `boring::ssl::SslStream` instead of `tokio-boring`? It seems like something that's generally useful, not just when using tokio.

Reply:
`boring::ssl::SslStream` felt more low-level to me, are you sure you want me to move it there?

Reply:
Done.